src/components/NewEventDialog.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import useForm from 'react-hook-form';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { TextField, Button, Grid, Box } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+
+import useStyle from './UseStyle';
+
+const mutation = gql`
+  mutation addEvent(
+    $title: String!
+    $start: String!
+    $end: String!
+    $description: String!
+  ) {
+    addEvent(
+      title: $title
+      start: $start
+      end: $end
+      description: $description
+    ) {
+      id
+      title
+      start
+      end
+      description
+    }
+  }
+`;
+
+function NewEventDialog(props) {
+  const { onClose, open, selectedDate } = props;
+  const classes = useStyle();
+
+  const { handleSubmit, register, errors } = useForm();
+  const [addEvent, { loading, error }] = useMutation(mutation);
+
+  const onSubmit = useCallback(
+    ({ title, description, date, startTime, endTime }) => {
+      addEvent({
+        variables: {
+          title,
+          start: new Date(date + ' ' + startTime).toISOString(),
```

Comment:
[MDN Date documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) discourages using the `Date` constructor from parsing dates: the behavior differs between browsers - this code works in Chrome, but breaks in Safari. It's possible to use a regexp here or a library such as [luxon](https://moment.github.io/luxon/docs/manual/parsing.html) which probably is future proof for this app which is dealing with dates.